foreign_cc/built_tools/cmake_build.bzl

```diff
@@ -1,16 +1,19 @@
 """ Rule for building CMake from sources. """
 
-load("//foreign_cc:defs.bzl", "configure_make")
+load("//foreign_cc:defs.bzl", "ninja")
 
 def cmake_tool(name, srcs, **kwargs):
     tags = ["manual"] + kwargs.pop("tags", [])
 
-    configure_make(
+    ninja(
         name = "{}.build".format(name),
-        configure_command = "bootstrap",
-        configure_options = ["--", "-DCMAKE_MAKE_PROGRAM=$$MAKE$$"],
-        # On macOS at least -DDEBUG gets set for a fastbuild
-        copts = ["-UDEBUG"],
+        env = {
+            "MAKE": "$(NINJA)",
+        },
+        tool_prefix = "$$EXT_BUILD_ROOT$$/external/cmake_src/bootstrap --generator=Ninja --prefix=$$INSTALLDIR -- -DCMAKE_MAKE_PROGRAM=$$MAKE && ",
```

Comment:
This is bit hacky but ninja doesn't currently have a configure / bootstrap command attribute available.